source/utilities.cc

```diff
@@ -1822,6 +1822,188 @@ namespace aspect
       return lookup->get_data(position,component);
     }
 
+
+
+    double
+    weighted_p_norm_average ( const std::vector<double> &weights,
+                              const std::vector<double> &parameter_values,
+                              const double p)
+    {
+      //TODO: prevent devision by zero for all
+      double averaged_parameter = 0.0;
+
+      // first look at the specal case which can be done faster
+      // Minimum
+      if (p <= -1000)
+        {
+          const unsigned int i = (unsigned int)(std::min_element( weights.begin(),
+                                                                  weights.end() )
+                                                - weights.begin());
+          averaged_parameter = parameter_values[i];
+          return averaged_parameter;
+        }
+      // Harmonic average (TODO: Check if we need to use a epsilon)
+      if (p == -1)
```

Comment:
But you *can* (and maybe should for clarify) write this as a sequence of `if ... else if ... else if ...` statements. The `else` part helps the reader understand that the previous `if` block isn't supposed to end up here.